### ISSUE TITLE
error after closing dismssing the page that has the map control

### ISSUE THREAD

**Issue by Pirate911** [amay077/Xamarin.Forms.GoogleMaps#281]
## BUG / PROBREM REPORT

i am having below issue since the last update 1.8.0

### DETAILS

  at Xamarin.Forms.GoogleMaps.iOS.MapRenderer.OnElementChanged (Xamarin.Forms.Platform.iOS.ElementChangedEventArgs`1[TElement] e) [0x00069] in <0bdc53fe1bce4b6e878e432cac02c7af>:0 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x00118] in <4cf9e9e173a04ea3aa841c775e6a23d7>:0 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].Dispose (System.Boolean disposing) [0x0007b] in <4cf9e9e173a04ea3aa841c775e6a23d7>:0 
  at Xamarin.Forms.Platform.iOS.ViewRenderer`2[TView,TNativeView].Dispose (System.Boolean disposing) [0x00000] in <4cf9e9e173a04ea3aa841c775e6a23d7>:0 
  at Xamarin.Forms.GoogleMaps.iOS.MapRenderer.Dispose (System.Boolean disposing) [0x000a8] in <0bdc53fe1bce4b6e878e432cac02c7af>:0 
  at Foundation.NSObject.Dispose () [0x00000] in /Users/builder/data/lanes/4466/a04678c2/source/xamarin-macios/src/Foundation/NSObject2.cs:133 
  at Xamarin.Forms.Platform.iOS.Platform.DisposeModelAndChildrenRenderers (Xamarin.Forms.Element view) [0x00038] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.iOS\Platform.cs:231 
  at Xamarin.Forms.Platform.iOS.Platform+<Xamarin-Forms-INavigation-PopModalAsync>d__23.MoveNext () [0x0015f] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.iOS\Platform.cs:132 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.8.0.175/src/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00047] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.8.0.175/src/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.8.0.175/src/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.8.0.175/src/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <54e129cad3ee4c61a6b46c3680ddc69a>:0 
  at Xamarin.Forms.Application+NavigationImpl+<OnPopModal>d__2.MoveNext () [0x00072] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Core\Application.cs:323 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.8.0.175/src/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00047] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.8.0.175/src/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.8.0.175/src/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.8.0.175/src/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <54e129cad3ee4c61a6b46c3680ddc69a>:0 
  at AmmanhApp.InspectorMapPage+<btnSave_Clicked>c__async1.MoveNext () [0x001aa] in /Users/pirate/Projects/xxxxApp/Views/Complain/MapPagexaml.cs:188 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.8.0.175/src/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<ThrowAsync>m__0 (System.Object state) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.8.0.175/src/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1018 
  at UIKit.UIKitSynchronizationContext+<Post>c__AnonStorey0.<>m__0 () [0x00000] in /Users/builder/data/lanes/4466/a04678c2/source/xamarin-macios/src/UIKit/UIKitSynchronizationContext.cs:24 
  at Foundation.NSAsyncActionDispatcher.Apply () [0x00000] in /Users/builder/data/lanes/4466/a04678c2/source/xamarin-macios/src/Foundation/NSAction.cs:163 
  at (wrapper managed-to-native) UIKit.UIApplication:UIApplicationMain (int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Users/builder/data/lanes/4466/a04678c2/source/xamarin-macios/src/UIKit/UIApplication.cs:79 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x00038] in /Users/builder/data/lanes/4466/a04678c2/source/xamarin-macios/src/UIKit/UIApplication.cs:63 
  at xxxxApp.iOS.Application.Main (System.String[] args) [0x0001c] in /Users/user1/Projects/xxxxApp/iOS/Main.cs:13 

### VERSIONS

* Xamarin.Forms.GoogleMaps - 1.8.0
* Xamarin.Forms - x.x.x
* 

### PLATFORMS

- [ ] Android
- [x] iOS
- [ ] UWP

### HOW TO REPRODUCE

(Let me show how to reproduce with steps or codes)

1. 
2. 
3. 

```csharp
// code for reproduce
` var animState = await map.AnimateCamera(CameraUpdateFactory.NewCameraPosition(
                    new CameraPosition(
                        pinTokyo.Position, // Tokyo
                        17d, // zoom
                        45d, // bearing(rotation)
                        60d)), // tilt
                    TimeSpan.FromSeconds(5));`

**Comment by deextercz** [comment]
Interesting during update from beto to 1.8.0 was updated just PolygonLogic (ToMutablePath, SetOnHolesChanged), added InfoWindowAnchor, OnSnapshot method.
Are you using any of this?

**Comment by jcesarmm** [comment]
Hello I-m getting this error in the update 1.8.0, the beta works good and it is only happening in ios

System.NullReferenceException: Object reference not set to an instance of an object

2017-04-25 17:27:52.939 SGT.Movil.Cliente.iOS[52513:270993] Unhandled managed exception:
Object reference not set to an instance of an object (System.NullReferenceException)
  at Xamarin.Forms.GoogleMaps.iOS.MapRenderer.OnElementChanged (Xamarin.Forms.Platform.iOS.ElementChangedEventArgs`1[TElement] e) [0x00069] in <0bdc53fe1bce4b6e878e432cac02c7af>:0 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x00110] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Platform.iOS\VisualElementRenderer.cs:167 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].Dispose (System.Boolean disposing) [0x0007b] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Platform.iOS\VisualElementRenderer.cs:218 
  at Xamarin.Forms.Platform.iOS.ViewRenderer`2[TView,TNativeView].Dispose (System.Boolean disposing) [0x00000] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Platform.iOS\ViewRenderer.cs:42 
  at Xamarin.Forms.GoogleMaps.iOS.MapRenderer.Dispose (System.B
oolean disposing) [0x000a8] in <0bdc53fe1bce4b6e878e432cac02c7af>:0 
  at Foundation.NSObject.Dispose () [0x00000] in /Users/builder/data/lanes/3985/ba11e481/source/xamarin-macios/src/Foundation/NSObject2.cs:133 
  at Xamarin.Forms.Platform.iOS.Platform.DisposeModelAndChildrenRenderers (Xamarin.Forms.Element view) [0x00038] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Platform.iOS\Platform.cs:231 
  at Xamarin.Forms.Platform.iOS.Platform.HandleChildRemoved (System.Object sender, Xamarin.Forms.ElementEventArgs e) [0x00007] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Platform.iOS\Platform.cs:342 
  at Xamarin.Forms.Element.OnDescendantRemoved (Xamarin.Forms.Element child) [0x00008] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Core\Element.cs:601 
  at Xamarin.Forms.Element.OnChildRemoved (Xamarin.Forms.Element child) [0x00021] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Core\Element.cs:368 
  at Xamarin.Forms.VisualElement.OnChildRemoved (Xamarin.Forms.Element child) [0x00000] 
in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Core\VisualElement.cs:578 
  at Xamarin.Forms.Page.OnInternalRemoved (Xamarin.Forms.VisualElement view) [0x00013] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Core\Page.cs:360 
  at Xamarin.Forms.Page.InternalChildrenOnCollectionChanged (System.Object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e) [0x00022] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Core\Page.cs:338 
  at System.Collections.ObjectModel.ObservableCollection`1[T].OnCollectionChanged (System.Collections.Specialized.NotifyCollectionChangedEventArgs e) [0x00012] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.4.0.128/src/mono/mcs/class/referencesource/System/compmod/system/collections/objectmodel/observablecollection.cs:288 
  at System.Collections.ObjectModel.ObservableCollection`1[T].OnCollectionChanged (System.Collections.Specialized.NotifyCollectionChangedAction action, System.Object item, System.Int32 index) [0x00000] in /Library/Framew
orks/Xamarin.iOS.framework/Versions/10.4.0.128/src/mono/mcs/class/referencesource/System/compmod/system/collections/objectmodel/observablecollection.cs:351 
  at System.Collections.ObjectModel.ObservableCollection`1[T].RemoveItem (System.Int32 index) [0x0002b] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.4.0.128/src/mono/mcs/class/referencesource/System/compmod/system/collections/objectmodel/observablecollection.cs:205 
  at System.Collections.ObjectModel.Collection`1[T].Remove (T item) [0x0002d] in /Library/Frameworks/Xamarin.iOS.framework/Versions/10.4.0.128/src/mono/mcs/class/referencesource/mscorlib/system/collections/objectmodel/collection.cs:113 
  at Xamarin.Forms.MasterDetailPage.set_Detail (Xamarin.Forms.Page value) [0x0004b] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Core\MasterDetailPage.cs:44 
  at SGT.Movil.Cliente.Pages.MenuPage.MasterPage_OpenSettingsPage (System.Object sender, System.EventArgs e) [0x00025] in D:\Project\ProyectoSGTGit\SGT.Movil\SGT.Movil.Cliente\SGT.Movil
.Cliente\Pages\MenuPage.xaml.cs:57 
  at SGT.Movil.Cliente.Pages.MenuPageMaster.TapGestureRecognizer_Tapped (System.Object sender, System.EventArgs e) [0x0000f] in D:\Project\ProyectoSGTGit\SGT.Movil\SGT.Movil.Cliente\SGT.Movil.Cliente\Pages\MenuPageMaster.xaml.cs:65 
  at Xamarin.Forms.TapGestureRecognizer.SendTapped (Xamarin.Forms.View sender) [0x0002e] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Core\TapGestureRecognizer.cs:46 
  at Xamarin.Forms.Platform.iOS.EventTracker+<>c__DisplayClass12_0.<GetNativeRecognizer>b__0 (UIKit.UITapGestureRecognizer r) [0x00045] in C:\BuildAgent\work\ca3766cfc22354a1\Xamarin.Forms.Platform.iOS\EventTracker.cs:94 
  at UIKit.UITapGestureRecognizer+Callback.Activated (UIKit.UITapGestureRecognizer sender) [0x00000] in /Users/builder/data/lanes/3985/ba11e481/source/xamarin-macios/src/UIKit/UIGestureRecognizer.cs:211 
--- End of stack trace from previous location where exception was thrown ---
  at (wrapper managed-to-native) UIKit.UIApplication:UIApplicationMain (int,s
tring[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Users/builder/data/lanes/3985/ba11e481/source/xamarin-macios/src/UIKit/UIApplication.cs:79 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x00038] in /Users/builder/data/lanes/3985/ba11e481/source/xamarin-macios/src/UIKit/UIApplication.cs:63 
  at SGT.Movil.Cliente.iOS.Application.Main (System.String[] args) [0x00001] in D:\Project\ProyectoSGTGit\SGT.Movil\SGT.Movil.Cliente\SGT.Movil.Cliente.iOS\Main.cs:17 
InspectorDebugSession(1): HandleTargetEvent: TargetExited
InspectorDebugSession(1): Disposed
The app has been terminated.
Launch failed. The app 'SGT.Movil.Cliente.iOS' could not be launched on 'iPhone 6 iOS 10.2'. Error: An error occurred while executing MTouch. Please check the logs for more details.
The app has been terminated.

**Comment by amay077** [comment]
Just released [v1.8.1](https://www.nuget.org/packages/Xamarin.Forms.GoogleMaps/1.8.1) what fixed it.

**Comment by Pirate911** [comment]
it's fixed . thank you so much .
